apps/prairielearn/src/middlewares/clientFingerprint.sql

```diff
@@ -0,0 +1,40 @@
+--BLOCK select_client_fingerprint
+SELECT
+  id, user_id, session_id, ip_address, user_agent, accept, accept_language
+FROM
+  client_fingerprints
+WHERE
+  user_id = $user_id
+  AND session_id = $session_id
+  AND ip_address = $ip_address
+  AND user_agent = $user_agent
+  AND accept = $accept
+  AND accept_language = $accept_language
+
+--BLOCK insert_client_fingerprint
+INSERT INTO client_fingerprints (
+  user_id,
+  session_id,
+  ip_address,
+  user_agent,
+  accept_language,
+  accept
+) 
+VALUES (
+  $user_id,
+  $session_id,
+  $ip_address,
+  $user_agent,
+  $accept_language,
+  $accept
+)
+RETURNING id
+
```

Comment:
As mentioned below, is there a better sql query we can write here to find and insert?